File: cs/XDust/XDust/XDustNotExistsNode.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace XDust
{
    public class XDustNotExistsNode : XDustExistsNode
    {
        public XDustNotExistsNode(String path, String scope, Dictionary<String, XDustNode> parameters)
            : base(path, scope, parameters)
        {
        }

        public override string ChooseBodyName(Context context, object model)
        {
            Object resolved = this.Context.Resolve(context, model);
            return this.IsTruthy(resolved) ? XDustLogicNode.ELSE : XDustLogicNode.BLOCK;
        }
    }
}

File: java/src/xdust-mvn/src/main/java/com/heydanno/xdust/XDustNotExistsNode.java
package com.heydanno.xdust;

import java.util.Map;

/**
 * Node for testing a data point and switching between two blocks based on its
 * falseyness
 */
public class XDustNotExistsNode extends XDustExistsNode {

	private static final long serialVersionUID = 6236854786516129400L;

	/**
	 * Constructor
	 * 
	 * @param path
	 *            The path to resolve against the data context
	 * @param scope
	 *            An alternate path to select the block's data context
	 * @param parameters
	 *            Variables to define for the block
	 */
	public XDustNotExistsNode(String path, String scope,
			Map<String, XDustNode> parameters) {
		super(path, scope, parameters);
	}

	/**
	 * Selects the name of the body to render
	 * 
	 * @param context
	 *            The current data context
	 * @param model
	 *            The current data model
	 * @return The name of the body to render
	 */
	@Override
	public String chooseBodyName(Context context, Object model) {
		Object resolved = this.getContext().resolve(context, model);
		return this.isTruthy(resolved) ? ELSE : BLOCK;
	}

}
